gubernator/main.py

```diff
@@ -191,30 +191,118 @@ def build_details(build_dir):
+    else:
+        dirs = [f.filename for f in gcs_ls('%s/artifacts' % build_dir)
+                if f.is_dir]
+        if not dirs:
+            path = re.sub(r'kubelet\.log', '', build_dir)
+            files = [f.filename for f in gcs_ls(path)]
+            if path + log_file in files:
+                return path + log_file
+            return ""
+        log_files = []
+        for folder in dirs:
+            filenames = [f.filename for f in gcs_ls(folder)]
+            if folder + log_file in filenames:
+                path = folder + log_file
+                if path in filenames:
+                    log_files.append(path)
+        return log_files
+
+
+@memcache_memoize('all-logs://', expires=60*60*4)
+def get_all_logs((directory, artifacts)):
+    log_files = {}
+    if artifacts:
+        dirs = [f.filename for f in gcs_ls('%s/artifacts' % directory)
+                if f.is_dir]
+    else:
+        dirs = [directory]
+    for d in dirs:
+        log_files[d] = []
+        for f in gcs_ls(d):
+            log_name = regex.log_re.search(f.filename)
+            if log_name:
+                log_files[d].append(f.filename)
+    return log_files
 
 
 def parse_log_file(log_filename, pod, filters=None, make_dict=False, objref_dict=None):
     """Based on make_dict, either returns the objref_dict or the parsed log file"""
     log = gcs_async.read(log_filename).get_result()
     if log is None:
-        return None
+        return None, False if make_dict else None
     pod_re = regex.wordRE(pod)
-
+    if objref_dict is None:
+        objref_dict = {}
     if make_dict:
-        return kubelet_parser.make_dict(log.decode('utf8', 'replace'), pod_re)
+        return kubelet_parser.make_dict(log.decode('utf8', 'replace'), pod_re, objref_dict)
     else:
         return log_parser.digest(log.decode('utf8', 'replace'),
         error_re=pod_re, filters=filters, objref_dict=objref_dict)
 
 
+def get_logs((build_dir, log_files, junit, pod_name, filters, objref_dict)):
+    # pylint: disable=too-many-locals
+    all_logs = {}
+    results = {}
+    apiserver_filename = find_log((build_dir, junit, "kube-apiserver.log"))
+    old_dict_len = len(objref_dict)
+    # For node e2e tests, the folder with the corresponding junit file is the only
+    # relevant one
+    if apiserver_filename and pod_name:
+        # default to filtering kube-apiserver log if user unchecks both checkboxes
+        if log_files == []:
+            log_files = [apiserver_filename]
+
+        artifact_filename = re.sub("/kube-apiserver.log", "", apiserver_filename)
+        all_logs = get_all_logs((artifact_filename, False))
+        parsed_dict, pod_in_file = parse_log_file("%s/kubelet.log" % artifact_filename,
+            pod_name, make_dict=True)
+        objref_dict.update(parsed_dict)
+        if log_files:
+            for log_file in log_files:
+                parsed_file = parse_log_file(log_file, pod_name, filters,
+                    objref_dict=objref_dict)
```

Comment:
What's the line-length limit for python in this repo? Unless it's 80 this probably shouldn't wrap...